src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipService.java

```diff
@@ -69,6 +71,27 @@ public String createLimitedPartnership(Transaction transaction,
         return insertedSubmission.getId();
     }
 
+    public void updateLimitedPartnership(String submissionId,
+                                         DataType type,
+                                         Map<String, Object> data) throws ServiceException {
+
+        var limitedPartnershipSubmissionDao = repository.findById(submissionId);
+
+        if (limitedPartnershipSubmissionDao.isEmpty()) {
+            throw new ServiceException(String.format(
+                    "Submission with id %s not found", submissionId));
+        }
+
+        if (type == DataType.EMAIL) {
+            DataDao dataDao = limitedPartnershipSubmissionDao.get().getData();
+
+            String email = (String) data.get("email");
```

Comment:
can "email" be replaced with DataType.EMAIL.getDescription() ?

Reply:
or better, type.getDescription()

Reply:
this line get the value send by the web not a type